app/src/org/commcare/google/services/analytics/FirebaseAnalyticsUtil.java

```diff
@@ -253,29 +258,38 @@ private static boolean rateLimitReporting(double percentOfEventsToReport) {
         return Math.random() < percentOfEventsToReport;
     }
 
-    public static void reportVideoPlayEvent(String videoName, long videoDuration, long videoStartTime) {
-        String videoUsage;
-        if (videoDuration == -1) {
-            videoUsage = VIDEO_USAGE_LENGTH_UNKNOWN;
+    private static String getVideoUsage(long totalDuration, long playDuration) {
+        if (totalDuration == -1) {
+            return VIDEO_USAGE_LENGTH_UNKNOWN;
         } else {
-
-            long timeSpend = new Date().getTime() - videoStartTime;
-            timeSpend = timeSpend + VIDEO_USAGE_ERROR_APPROXIMATION;
-            double timeSpendInFraction = timeSpend / videoDuration;
+            double timeSpendInFraction = playDuration / totalDuration;
 
             if (timeSpendInFraction <= 0.25) {
-                videoUsage = VIDEO_USAGE_IMMEDIATE;
+                return VIDEO_USAGE_IMMEDIATE;
             } else if (timeSpendInFraction <= 0.5) {
-                videoUsage = VIDEO_USAGE_PARTIAL;
+                return VIDEO_USAGE_PARTIAL;
             } else if (timeSpendInFraction <= 0.75) {
-                videoUsage = VIDEO_USAGE_MOST;
+                return VIDEO_USAGE_MOST;
             } else if (timeSpendInFraction <= 2) {
-                videoUsage = VIDEO_USAGE_FULL;
+                return VIDEO_USAGE_FULL;
             } else {
-                videoUsage = VIDEO_USAGE_OTHER;
+                return VIDEO_USAGE_OTHER;
             }
         }
+    }
+
+    public static void reportVideoPlayEvent(String videoName, long videoDuration, long videoStartTime) {
+        long timeSpend = new Date().getTime() - videoStartTime;
+        timeSpend = timeSpend + VIDEO_USAGE_ERROR_APPROXIMATION;
```

Comment:
The `VIDEO_USAGE_ERROR_APPROXIMATION` should only be added for external videos and not internal since there is no delay in user clicking the play and video start in case of internal videos.